Demo/Common/IntegrationExamples/Components/CardComponentAdvancedFlowExample.swift

```diff
@@ -0,0 +1,200 @@
+//
+// Copyright (c) 2022 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Adyen
+import AdyenActions
+import AdyenCard
+import AdyenComponents
+import AdyenDropIn
+import PassKit
+import UIKit
+
+internal final class CardComponentAdvancedFlowExample: InitialDataAdvancedFlowProtocol {
+
+    internal var paymentMethods: PaymentMethods?
+    internal var cardComponent: PresentableComponent?
+    internal weak var presenter: PresenterExampleProtocol?
+
+    // MARK: - Action Handling
+
+    internal lazy var adyenActionComponent: AdyenActionComponent = {
+        let handler = AdyenActionComponent(context: context)
+        handler.configuration.threeDS.delegateAuthentication = ConfigurationConstants.delegatedAuthenticationConfigurations
+        handler.configuration.threeDS.requestorAppURL = URL(string: ConfigurationConstants.returnUrl)
+        handler.delegate = self
+        handler.presentationDelegate = self
+        return handler
+    }()
+
+    // MARK: - Initializers
+
+    internal init() {}
+
+    // MARK: - Networking
+
+    internal func requestInitialData(completion: ((PaymentMethods?, Error?) -> Void)?) {
+        requestPaymentMethods(order: nil) { [weak self] paymentMethods, errorResponse in
+            guard paymentMethods != nil else {
+                guard let errorResponse = errorResponse else {
+                    return
+                }
+                self?.presenter?.presentAlert(with: errorResponse, retryHandler: {
+                    self?.requestPaymentMethods(order: nil, completion: completion)
+                })
+                return
+            }
+            self?.paymentMethods = paymentMethods
+        }
+    }
+
+    internal func presentCardComponent() {
+        guard let component = cardComponent(from: paymentMethods) else { return }
+        component.cardComponentDelegate = self
+        present(component, delegate: self)
+    }
+
+    internal func cardComponent(from paymentMethods: PaymentMethods?) -> CardComponent? {
+        guard let paymentMethods = paymentMethods,
+              let paymentMethod = paymentMethods.paymentMethod(ofType: CardPaymentMethod.self) else { return nil }
+        let style = FormComponentStyle()
+        let config = CardComponent.Configuration(style: style)
+        return CardComponent(paymentMethod: paymentMethod,
+                             context: context,
+                             configuration: config)
+    }
+
+    // MARK: - Presentation
+
+    private func present(_ component: PresentableComponent, delegate: (PaymentComponentDelegate & ActionComponentDelegate)) {
```

Comment:
This function was a common one to combine both payment and action components. here it is not needed since there is only one component. 

And inside no need to check for `ActionComponent`

Reply:
It is needed: used on [line 199](https://github.com/Adyen/adyen-ios/pull/1199/files#diff-88c1e7bf4910e94695aab972b93a1876ba1de6e2c79adc181cba06796a572d3eR198)

Reply:
It can be simplified and moved to `PresenterExampleProtocol`
See [outdated comment](https://github.com/Adyen/adyen-ios/pull/1199#discussion_r1173601171)

Reply:
e7c03971d28f7f4fb10a23fe39580f2dacb88ac4